File: data-pipeline/checksum-generator/src/test/java/com/library/checksum/fixtures/Event.java
package com.library.checksum.fixtures;

import com.library.checksum.system.Mappable;

import java.util.Map;


public class Event implements Mappable {

    private Map<String,Object> map;

    public Event(Map<String,Object> map){
        this.map=map;
    }

    @Override
    public Map<String, Object> getMap() {
        return this.map;
    }

    @Override
    public void setMetadata(Map<String, Object> metadata) {
        Map<String,Object> eventMap = getMap();
        eventMap.put("metadata",metadata);
    }
}

File: data-pipeline-flink/assessment-aggregator/src/main/scala/org/sunbird/dp/assessment/domain/Event.scala
package org.sunbird.dp.assessment.domain

import java.util
import org.sunbird.dp.core.domain.{Events, EventsPath}

class Event(eventMap: util.Map[String, Any]) extends Events(eventMap) {

  private val jobName = "AssessmentAggregator"

  def assessmentEts: Long = {
    val ets = telemetry.read[Long]("assessmentTs").get
    ets.longValue()

  }

  def courseId: String = {
   telemetry.read[String]("courseId").get
  }

  def contentId: String = {
    telemetry.read[String]("contentId").get
  }

  def batchId: String = {
    telemetry.read[String]("batchId").get
  }

  def attemptId: String = {
    telemetry.read[String]("attemptId").get
  }

  def userId: String = {
    telemetry.read[String]("userId").get
  }

  def assessEvents: util.ArrayList[util.Map[String, AnyRef]] = {
    telemetry.read[util.ArrayList[util.Map[String, AnyRef]]]("events").getOrElse(null)
  }

  def markFailed(errorMsg: String): Unit = {
    telemetry.addFieldIfAbsent(EventsPath.FLAGS_PATH, new util.HashMap[String, Boolean])
    telemetry.addFieldIfAbsent("metadata", new util.HashMap[String, AnyRef])
    telemetry.add("metadata.validation_error", errorMsg)
    telemetry.add("metadata.src", jobName)
  }

}
